File: backend/components/metiers/threed/api/samples/middlewares/sample_middlewares.js
/* global console */
// Exemple ultra avancé clé en main des middlewares API Threed (JS)
const { auditRequest, rgpdMiddleware, accessibilityMiddleware } = require('../middlewares/middlewares');

function sampleMiddlewaresUltra(app) {
  app.use(rgpdMiddleware);
  app.use(accessibilityMiddleware);
  app.use(auditRequest);
  console.info('Middlewares ultra avancé appliqués');
  console.log('Middlewares ultra avancé exécuté avec succès.');
}

module.exports = { sampleMiddlewaresUltra };

File: backend/components/metiers/threed/api/samples/middlewares/sample_middlewares.py
# Exemple ultra avancé clé en main des middlewares API Threed (Python)
from ..middlewares.middlewares import audit_request, rgpd_middleware, accessibility_middleware
from fastapi import FastAPI
import logging

def sample_middlewares_ultra():
    app = FastAPI()
    app.middleware('http')(rgpd_middleware)
    app.middleware('http')(accessibility_middleware)
    app.middleware('http')(audit_request)
    logging.info('Middlewares ultra avancé appliqués')
    print('Middlewares ultra avancé exécuté avec succès.')
